src/cart/crt_context.c

```diff
@@ -1516,8 +1521,12 @@ crt_context_req_untrack(struct crt_rpc_priv *rpc_priv)
 			epi->epi_req_num++;
 			D_ASSERT(epi->epi_req_num >= epi->epi_reply_num);
 
-			/* add to resend list */
-			d_list_add_tail(&tmp_rpc->crp_tmp_link, &submit_list);
+			/* add to submit list if not cancelled or timed out already  */
+			if (tmp_rpc->crp_timeout_ts != 0) {
```

Comment:
[Request for changes] Would `crp_timeout_ts` ever be 0 after the `crt_set_timeout` call above on new line 1509? Should we instead do this check at the very beginning, that is, new line 1508?

Reply:
good catch